### PR TITLE
Che in Che stack improvements.

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
Upgrade memory limit to 7G = 5G to build che +2 G for ws-agent
- used centos based image
- upgrade of maven to 3.5
- upgrade of tomcat to 8.5.23

Che build tested with command
```
docker run -it --rm --name build-che \
           -v "$HOME/hdd/docker-m2/.m2:/home/user/.m2"  \
           -v "$PWD":/home/user/che-build  \
           -w /home/user/che-build  \
           -e "JAVA_OPTS=-Xms256m -Xmx2048m -Djava.security.egd=file:/dev/./urandom" \
           -e "MAVEN_OPTS=-Xms256m -Xmx2048m -Djava.security.egd=file:/dev/./urandom" \
           -m 5g \
           --memory-swappiness 0 \
           eclipse/che-dev:nightly  \
           mvn  -Dgwt.compiler.localWorkers=2  \
               -DskipIntegrationTests=true \
               -Pnative \
               clean install 

```
### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/6928

#### Release Notes
Used Centos based image for che in che stack

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
maven is not doing HTTPS by default now ?

**Comment by benoitf** [comment]
https://maven.apache.org/docs/3.2.3/release-notes.html

**Comment by benoitf** [comment]
so I could say settings.xml is useless

File: dockerfiles/dev/Dockerfile [hunk 1]
```diff
@@ -6,69 +6,50 @@
...
+           make \
+           golang \
+    sudo yum clean all && \
+    cat /opt/rh/rh-maven35/enable >> /home/user/.bashrc  && \
+    cat /opt/rh/rh-nodejs6/enable >> /home/user/.bashrc && \
+    sudo ln -s /opt/rh/rh-nodejs6/root/usr/bin/node /usr/local/bin/nodejs
+
+RUN mkdir $HOME/.m2 && \
+    mkdir /home/user/tomcat8 && \
+    wget -qO- "https://archive.apache.org/dist/tomcat/tomcat-8/v8.5.23/bin/apache-tomcat-8.5.23.tar.gz" | tar -zx --strip-components=1 -C /home/user/tomcat8 && \
```
Comment (benoitf) [inline dockerfiles/dev/Dockerfile]: It's not a env var ? For the version. It's easier to update

File: dockerfiles/dev/Dockerfile [hunk 2]
```diff
@@ -6,69 +6,50 @@
...
-    && sudo rm "node-v$NODE_VERSION-linux-x64.tar.gz" SHASUMS256.txt.asc
+FROM registry.centos.org/che-stacks/centos-stack-base
+
+EXPOSE 4403 8080 8000 9876 22
+
+RUN sudo yum -y update && \
+    sudo yum -y install  \
+           rh-maven35 \
+           plexus-classworlds \
+           rh-nodejs6 \
```
Comment (benoitf) [inline dockerfiles/dev/Dockerfile:19]: If we need nodejs shouldn't we extend nodejs image ? Instead of base ?

**Comment by skabashnyuk** [comment]
settings.xml  has been removed

**Review by benoitf — APPROVED** [review]

File: dockerfiles/dev/Dockerfile (hunk 2 above)
Comment (skabashnyuk) [reply dockerfiles/dev/Dockerfile:19]: I've tried but I was not able to build che with it outside from che.
with command
```
docker run -it --rm --name build-che \
           -v "$HOME/hdd/docker-m2/.m2:/home/user/.m2"  \
           -v "$PWD":/home/user/che-build  \
           -w /home/user/che-build  \
           -e "JAVA_OPTS=-Xms256m -Xmx2048m -Djava.security.egd=file:/dev/./urandom" \
           -e "MAVEN_OPTS=-Xms256m -Xmx2048m -Djava.security.egd=file:/dev/./urandom" \
           -m 5g \
           --memory-swappiness 0 \
           eclipse/che-dev:nightly  \
           mvn  -Dgwt.compiler.localWorkers=2  \
               -DskipIntegrationTests=true \
               -Pnative \
               clean install 
```
I had ```node not found``` error

**Review by ghost — APPROVED** [review]

**Pull request merged**